### PR TITLE
Return False for is_flagged(anonymous_user)

### PR DESCRIPTION
**Pull request by ghost**
explained in #12

### CODE REVIEW

File: siteflags/models.py [hunk 1]
```diff
@@ -314,6 +314,9 @@ def is_flagged(self, user: 'User' = None, *, status: int = None) -> int:
         :param status: Optional status filter
 
         """
+        if user and not user.id:
```
Comment (idlesign) [inline siteflags/models.py]: `is_anonymous` check would be more obvious

**Review by idlesign — CHANGES_REQUESTED** [review]

**Comment by idlesign** [comment]
Thank you. There's a note though.

**Comment by idlesign** [comment]
A test case would be nice too.

File: siteflags/models.py (hunk 1 above)
Comment (BoPeng) [reply siteflags/models.py]: I used `not user.id` because that was the one used for `set_flag`.

**Comment by BoPeng** [comment]
I did not add a test because `pytest-django` does not have a fixture directly for anonymous user. Do you have an easy way to create an anonymous user?

I guess it is easiest just create an `AnonymousUser` instance from `django.contrib.auth.models`.

**Comment by bioworkflows** [comment]
Used `is_anonymous` and added a test.

**Comment by idlesign** [comment]
> I did not add a test because `pytest-django` does not have a fixture directly for anonymous user.

But this project doesn't use `pytest-django`.
`pytest-djangoapp` has `user_create` fixture which accepts `anonymous` param: https://pytest-djangoapp.readthedocs.io/en/latest/fixtures_users.html#pytest_djangoapp.fixtures.users.user_create

**Comment by BoPeng** [comment]
Revised to use `user_create(anaonymous=True)`. This PR should be ready to merge.

**Comment by idlesign** [comment]
Thank you. Merged.

**Pull request merged**